src/hooks/useToggle.ts

```diff
@@ -0,0 +1,11 @@
+import React from "react"
+
+// Taken from this snippet:
+// https://www.joshwcomeau.com/snippets/react-hooks/use-toggle/
+export default function useToggle(initialValue = false): [boolean, () => void] {
```

Comment:
IMO this results in things just being a little bit more confusing. This is a pretty simple snippet and I would personally prefer having it in the component.

Reply:
I understand it seems like overkill in this one scenario, but I think being able to encapsulate even simple patterns like this into custom hooks/helpers will keep complexity and potential bugs out of component code.

Reply:
Do we think this will be used in more components? If so I think it makes sense to keep it in this separate file. Otherwise I think we can keep it in the component and then pull it out when need be. However if we are keeping it separate, I would suggest adding a bit more description to why this hook is needed. Josh Comeau has a good explanation but it would be nice to get a shortened version of that in this file so we don't need to click on the link